serde/src/private/de.rs

```diff
@@ -525,12 +525,17 @@ mod content {
             Ok(Content::Map(vec))
         }
 
-        fn visit_enum<V>(self, _visitor: V) -> Result<Self::Value, V::Error>
+        fn visit_enum<V>(self, visitor: V) -> Result<Self::Value, V::Error>
         where
             V: EnumAccess<'de>,
         {
-            Err(de::Error::custom(
-                "untagged and internally tagged enums do not support enum input",
+            let (key, data) = tri!(visitor.variant::<String>());
+            Ok(Content::Map(
+                [(
+                    Content::String(key),
+                    tri!(data.newtype_variant::<Self::Value>()),
+                )]
+                .into(),
```

Comment:
It seems that you should at least use `Content` instead of `String`, and probably introduce a new variant in `Content` for enum.

Also, need an explanation, why you choose `newtype_variant`? If that the only thing that expected when deserializing untagged and internally tagged enums, then this should be explicitly explained, because this is nor obvious at all.

The last, but not least, you should add tests that cover this change. I tried to bring some hierarchial structure to the serde tests so that the new contributors know where to start, but not all PRs are accepted yet. However, for that part tests seems, already concentrated in one place:
- [untagged](https://github.com/serde-rs/serde/blob/b9dbfcb4ac3b7a663d9efc6eb1387c62302a6fb4/test_suite/tests/test_enum_untagged.rs)
- [internally tagged](https://github.com/serde-rs/serde/blob/b9dbfcb4ac3b7a663d9efc6eb1387c62302a6fb4/test_suite/tests/test_enum_internally_tagged.rs)

I think that adding those details will increase chance of the PR to be reviewed by maintainers.